src/tabmat/categorical_matrix.py

```diff
@@ -422,6 +458,11 @@ def tocsr(self) -> sps.csr_matrix:
         # TODO: data should be uint8
         data = np.ones(self.shape[0], dtype=int)
 
+        if self.drop_first:
+            return sps.csr_matrix(
+                (data, self.indices, np.arange(self.shape[0] + 1, dtype=int)),
+                shape=(self.shape[0], self.shape[1] + 1),
```

Comment:
```suggestion
                shape=(self.shape[0], self.shape[1] + self.drop_first),
```
and collapse the two blocks?

Reply:
This was done on purpose. If you look closely, the `drop_first` block also has a slicing operation at the end. This is inefficient because it always copies the data, even if you select everything. Therefore, only the `drop_first` version has this inefficiency.